config/config.py

```diff
@@ -17,7 +17,12 @@ def signFilter(poi):
         sign_filter = os.environ['RENDER_SIGNS_FILTER']
         hide_filter = os.environ['RENDER_SIGNS_HIDE_FILTER'] == 'true'
         # Transform the lines into an array and strip whitespace from each line.
-        lines = list(map(lambda l: l.strip(),[poi['Text1'], poi['Text2'], poi['Text3'], poi['Text4']]))
+        lines = list(map(lambda l: l.strip(), [poi['Text1'], poi['Text2'], poi['Text3'], poi['Text4']]))
+        # Remove all leading and trailing empty lines
+        while lines and not lines[0]:
+            del lines[0]
+        while lines and not lines[-1]:
+            del lines[-1]
```

Comment:
Instead of modifying the list, maybe we could create a utility function that returns a slice of the list? It's not a big deal though since the list being modified is created just before this.

Reply:
There is something funky with how Overviewer is reading this file; it doesn't seem to trivially support additional functions. I can look further into it (normally I don't like to jam so much into a function but I found it failed before).

Reply:
Very weird! I don't think it's a blocker right now, but we should probably look into why that happens in case it becomes a blocker in the future.

Reply:
After some digging, I think it's because `minecraft-overviewer` upstream may use [`pickle`](https://docs.python.org/3/library/pickle.html) and it may not be configured to bring along additional functions. I am looking into potential workarounds.

[`genPOI.py` in `github.com/overviewer/Minecraft-Overviewer`](https://github.com/overviewer/Minecraft-Overviewer/blob/dc6f8cfcbd1056facac369152e88fb24bab86cc9/overviewer_core/aux_files/genPOI.py#L91)

----

I tested by adding the following to my render config:
```
def foo():
    return True
...
def signFilter(poi):
    foo()
```
and I got the following error:
```
2020-04-26 16:02:04 E An error has occurred. This may be a bug. Please let us know!
See http://docs.overviewer.org/en/latest/index.html#help

This is the error that occurred:
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/usr/lib/python3.5/dist-packages/overviewer_core/aux_files/genPOI.py", line 131, in parseBucketChunks
    result = ff(poi)
  File "/home/minecraft/config.py", line 17, in signFilter
    foo()
NameError: name 'foo' is not defined
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/bin/overviewer.py", line 650, in <module>
    ret = main()
  File "/usr/bin/overviewer.py", line 168, in main
    g.genPOI.main()
  File "/usr/lib/python3.5/dist-packages/overviewer_core/aux_files/genPOI.py", line 569, in main
    handleEntities(rset, config, args.config, list(rset_filters), markers)
  File "/usr/lib/python3.5/dist-packages/overviewer_core/aux_files/genPOI.py", line 211, in handleEntities
    results = pool.map(parseBucketChunks, ((buck, rset, filters) for buck in buckets))
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 260, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 608, in get
    raise self._value
NameError: name 'foo' is not defined
```

Reply:
Looks like this could be related https://github.com/overviewer/Minecraft-Overviewer/issues/1130#issuecomment-50502322 (Still an open issue as of 4/26/2020).

Reply:
I'll add this as a To-Do Cleanup and merge this as-is for now.